bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/types/BUnionType.java

```diff
@@ -117,4 +117,15 @@ public int hashCode() {
     public boolean isNilable() {
         return nullable;
     }
+
+    private boolean checkNillable(List<BType> memberTypes) {
+        boolean nillable = false;
+        for (BType t : memberTypes) {
+            if (t.isNilable()) {
+                nillable = true;
+                break;
```

Comment:
We can slightly improve this code by directly returning true or false. So we do not need temp nillable var.